test.sh

```diff
@@ -1,36 +1,133 @@
 #!/usr/bin/env bash
+set -ex
 
 function finish {
   echo '-----------------------test.sh------------------------------'
   echo 'Removing test environment'
   echo '------------------------------------------------------------'
   docker-compose down -v
 }
+
 trap finish EXIT
 
 function main() {
-  prepareOutputDir
-  createTestEnvironment
-  loadTestPolicy
-  runTests
-  printProxyConfiguration
-  initializeCert
-  runHTTPSTests
+  runDAP
+  runOSS
+}
+
+# Run DAP Enterprise test suite
+function runDAP() {
+  createDAPTestEnvironment
+  loadDAPTestPolicy
+  initializeDAPCert
+  runDAPTests
+}
+
+# Run OSS test suite
+function runOSS () {
+  createOSSTestEnvironment
+  loadOSSTestPolicy
+  runOSSTests
+  printOSSProxyConfiguration
+  initializeOSSCert
+  runOSSHTTPSTests
+}
+
+# Build DAP test container & start the cluster
+function createDAPTestEnvironment() {
+  docker-compose build --pull client cuke-master test-dap
+  export CONJUR_APPLIANCE_URL="https://cuke-master"
+  docker-compose up -d client cuke-master test-dap
+
+  # Delay to allow time for conjur to come up
+  echo 'Waiting for conjur server to be healthy'
+  docker-compose run --rm test ./wait_for_server.sh
+}
+
+function loadDAPTestPolicy() {
+  echo '-----------------------test.sh------------------------------'
+  echo "Loading DAP test policy"
+  echo '------------------------------------------------------------'
+
+  dap_client_cid=$(docker-compose ps -q client)
+
+  # Get certificate from cuke-master
+  ssl_cert=$(docker-compose exec -T cuke-master cat /opt/conjur/etc/ssl/conjur.pem)
+
+  docker exec \
+    -e CONJUR_SSL_CERTIFICATE="$ssl_cert" \
+    ${dap_client_cid} conjur authn login -u admin -p secret
+
+  # copy test-policy into a /tmp/test-policy within the client container
+  docker cp test-policy ${dap_client_cid}:/tmp
+
+  docker exec \
+    -e CONJUR_SSL_CERTIFICATE="$ssl_cert" \
+    ${dap_client_cid} conjur policy load root /tmp/test-policy/root.yml
+}
+
+function initializeDAPCert() {
+  echo '-----------------------test.sh------------------------------'
+  echo "Fetch certificate for DAP using client cli"
+  echo '------------------------------------------------------------'
+
+  dap_client_cid=$(docker-compose ps -q client)
+
+  # Get the pem file from conjur server
+  CONJUR_ACCOUNT="cucumber"
+  CONJUR_PROXY="https://cuke-master"
+
+  echo "remove old pem file"
+  rm -rf /test-cert/*
+
+  echo "fetch pem file from enterprise server"
+  exec_command="echo yes | conjur init -u '${CONJUR_PROXY}' -a '${CONJUR_ACCOUNT}'"
+  docker exec ${dap_client_cid} /bin/bash -c "$exec_command"
+
+  echo "convert pem to der file and copy it to share memory"
+  convert_command="openssl x509 -outform der -in /root/conjur-cucumber.pem -out /test-cert/conjur-cucumber.der"
```

Comment:
It would be nice to split this command into multiple lines as well since it's too long